homeassistant/components/tplink/light.py

```diff
@@ -59,14 +63,14 @@ def __init__(
         """Initialize the switch."""
         super().__init__(device, coordinator)
         # For backwards compat with pyHS100
-        if self.device.is_dimmer:
+        if device.is_dimmer:
             # Dimmers used to use the switch format since
             # pyHS100 treated them as SmartPlug but the old code
             # created them as lights
             # https://github.com/home-assistant/core/blob/2021.9.7/homeassistant/components/tplink/common.py#L86
             self._attr_unique_id = legacy_device_id(device)
```

Comment:
Is there some specific point when we should drop the legacy device id?

Reply:
We would need to do a migration of entity ids